geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/ProtobufSerializationService.java

```diff
@@ -41,12 +51,18 @@ public ProtobufSerializationService() {}
    */
   @Override
   public BasicTypes.EncodedValue encode(Object value) throws EncodingException {
+    BasicTypes.EncodedValue.Builder builder = BasicTypes.EncodedValue.newBuilder();
+
     if (value == null) {
-      return BasicTypes.EncodedValue.newBuilder().setNullResult(NullValue.NULL_VALUE).build();
+      return builder.setNullResult(NullValue.NULL_VALUE).build();
     }
 
-    BasicTypes.EncodedValue.Builder builder = BasicTypes.EncodedValue.newBuilder();
     try {
+      ByteString encoded = serializer.serialize(value);
```

Comment:
I was thinking that people might want to override the serialization of primitives, which this would let them do. They have the option of returning null and telling us to handle primitives instead.

Reply:
If we do this for primitive types we need to document which types we support.  Also, javadocs say we'll try to convert non-primitives to JSON but that's not true - we only convert PdxInstances into JSON.  Anything else will cause an exception to be thrown.